src/main/java/cn/nukkit/Player.java

```diff
@@ -3816,8 +3816,7 @@ public void kill() {
         }
 
         PlayerDeathEvent ev = new PlayerDeathEvent(this, this.getDrops(), new TranslationContainer(message, params.toArray(new String[0])), this.expLevel);
-        ev.setKeepExperience(this.level.gameRules.getBoolean(GameRule.KEEP_INVENTORY));
-        ev.setKeepInventory(ev.getKeepExperience());
+        ev.setKeepInventory(this.level.gameRules.getBoolean(GameRule.KEEP_INVENTORY));
```

Comment:
This shouldn't be a problem because it's set before the event is called. Also `setKeepExperience` is first set to the keep inventory gamerule and `setKeepInventory` to `getKeepExperience` so both are set to the correct value which is same as the gamerule.